src/onefetch/deps.rs

```diff
@@ -0,0 +1,122 @@
+use {
+    crate::onefetch::error::*,
+    std::collections::HashMap,
+    std::{ffi::OsStr, fs},
+};
+
+pub enum PackageManager {
+    Cargo,
+    GoModules,
+    Npm,
+    Pip,
+}
+
+impl std::fmt::Display for PackageManager {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            PackageManager::Cargo => write!(f, "Cargo"),
+            PackageManager::GoModules => write!(f, "Go Modules"),
+            PackageManager::Npm => {
+                if std::path::Path::new("./yarn.lock").exists() {
+                    write!(f, "Yarn")
+                } else {
+                    write!(f, "Npm")
+                }
+            }
+            PackageManager::Pip => write!(f, "Pip"),
+        }
+    }
+}
+
+type DependencyParser = fn(&str) -> Option<i32>;
+
+pub struct Detector {
+    package_managers: HashMap<String, (DependencyParser, PackageManager)>,
+}
+
+// Package parsers go here. Parsers should take stirng contents and output a i32
+mod package_parsers {
+    use {regex::Regex, toml::Value};
+
+    pub fn cargo(contents: &str) -> Option<i32> {
+        let parsed = contents.parse::<Value>().unwrap();
+
+        Some(parsed["dependencies"].as_table().unwrap().len() as i32)
+    }
+
+    pub fn gomodules(contents: &str) -> Option<i32> {
+        let count = Regex::new(r"v[0-9]+").unwrap().find_iter(contents).count();
+
+        Some(count as i32)
+    }
+
+    pub fn npm(contents: &str) -> Option<i32> {
+        let parsed = json::parse(contents).unwrap();
+
+        Some(parsed["dependencies"].len() as i32)
+    }
+
+    pub fn pip(contents: &str) -> Option<i32> {
+        let count = Regex::new(r"(^[A-z]+)|(\n[A-z]+)").unwrap().find_iter(contents).count();
```

Comment:
This looks OK to me, but you might want to iterate over `contents.lines()`, filter to lines matching a regex, then return the count. This allows for a simpler regex, since you only have to check 1 line.

Reply:
Yeah I'll do that. I didn't know that existed

Reply:
Could we just do `(^|\n)[A-z]+`?

Reply:
I changed it in a19ced8 to `(^|\n)[A-z]+`. It seems easier to just do it that way

Reply:
That works. [You can also build it as a multi-line regex](https://docs.rs/regex/1.4.2/regex/struct.RegexBuilder.html#method.multi_line) if you want.